obspy/core/tests/test_event.py

```diff
@@ -754,11 +782,47 @@ def test_adding_a_referred_object_after_creation(self):
         self.assertEqual(ref_b.get_referred_object(), None)
         self.assertEqual(ref_c.get_referred_object(), None)
         # Setting the object for one will make it available to all other
-        # instances.
+        # instances, provided they weren't bound to specific objects.
         ref_b.set_referred_object(obj)
-        self.assertEqual(id(ref_a.get_referred_object()), obj_id)
-        self.assertEqual(id(ref_b.get_referred_object()), obj_id)
-        self.assertEqual(id(ref_c.get_referred_object()), obj_id)
+        self.assertIs(ref_a.get_referred_object(), obj)
+        self.assertIs(ref_b.get_referred_object(), obj)
+        self.assertIs(ref_c.get_referred_object(), obj)
+
+    def test_getting_gc_no_shared_resource_id(self):
+        """
+        Test that calling get_referred_object on a resource id whose object
+        has been garbage collected, and whose resource_id is unique,
+        returns None
+        """
+        obj1 = UTCDateTime()
+        rid1 = ResourceIdentifier(referred_object=obj1)
+        # delete obj1, make sure rid1 return None
+        del obj1
+        self.assertIs(rid1.get_referred_object(), None)
+
+    def test_getting_gc_with_shared_resource_id(self):
+        """
+        Test that calling get_referred_object on a resource id whose object
+        has been garbage collected, but that has another object that shares
+        the same resource_id, returns the other object with the same resource
+        id and issues a warning
+        """
+        uri = 'testuri'
+        obj1 = UTCDateTime()
+        obj2 = UTCDateTime()
+        rid1 = ResourceIdentifier(uri, referred_object=obj1)
+        rid2 = ResourceIdentifier(uri, referred_object=obj2)
+        self.assertFalse(rid1.get_referred_object() is
+                         rid2.get_referred_object())
+        del obj1
+        warnings.simplefilter('default')
+        with warnings.catch_warnings(record=True) as w:
+            rid1.get_referred_object()
+            assert len(w)
```

Comment:
Can you please add some fuzzy test of the warning message? Just to make sure that the correct warning is actually caught here.